### PR TITLE
MSI: Add checksums for executable files

### PR DESCRIPTION
**Pull request by ikreb7**

### CODE REVIEW

**Comment by mattock** [comment]
Looks reasonable, but I don't know much about MSI. The Travis-CI build failure is completely unrelated to this.

@rozmansi thoughts on this PR?

**Comment by rozmansi** [comment]
Depends on the compiler. MSVC always populates the checksum field of the PE image in output DLL/EXE. However, I am not sure about mingw.

Important: If the entry in the File table in MSI has `msidbFileAttributesChecksum` flag set saying the file _does contain_ a checksum, but the file _really does not_, the MSI will panic on installation.

@ikreb7 did you test the installer with mingw-built OpenVPN binaries?

The purpose of `msidbFileAttributesChecksum` flag is to mark which files can be "repaired by comparing checksums" using `msiexec /f`.

File: windows-msi/msi.wxs [hunk 1]
```diff
@@ -721,28 +721,28 @@
 
                     <Directory Id="BINDIR" Name="bin">
                         <Component Id="bin.libcrypto_1_1.dll" Guid="{811694A9-DC99-4CFA-BD72-4877F8E32C7C}">
-                            <File Name="libcrypto-1_1$(var.OPENSSL_PLAT).dll" Source="!(bindpath.openvpn)bin\libcrypto-1_1$(var.OPENSSL_PLAT).dll"/>
+                            <File Name="libcrypto-1_1$(var.OPENSSL_PLAT).dll" Source="!(bindpath.openvpn)bin\libcrypto-1_1$(var.OPENSSL_PLAT).dll Checksum="yes""/>
```
Comment (rozmansi) [inline windows-msi/msi.wxs:724]: This line contains a syntax error:

> error CNDL0104 : Not a valid source file; detail: 'yes' is an unexpected token. Expecting white space. Line 724, position 157.

Fix quoting, please.

**Comment by rozmansi** [comment]
Compiled openvpn using mingw; manually fixed the quoting in msi.wxs; tried to build the MSI packages, failed for missing openvpn.8.html; investigated; installed python-docutils; rebuilt the openvpn; built the MSI packages; tested them on a vanilla Windows box to confirm the checksums work.

File: windows-msi/msi.wxs (hunk 1 above)
Comment (ikreb7) [reply windows-msi/msi.wxs]: Fixed with 71ca738.

**Comment by rozmansi** [comment]
@ikreb7 thanks. Also thanks for fixing the Travis issue with openvpn.8.html and missing python-docutils.

Now the Travis fails later in the process. But, that's not related to this PR.

So, Acked-by: @rozmansi

File: windows-msi/msi.wxs (hunk 1 above)
Comment (lstipakov) [reply windows-msi/msi.wxs]: @rozmansi @ikreb7 are you sure that `Checksum` should be part of `Source`? I still got the same error. Shouldn't the proper fix be:

```
diff --git a/windows-msi/msi.wxs b/windows-msi/msi.wxs
index 27aabba..6604d73 100644
--- a/windows-msi/msi.wxs
+++ b/windows-msi/msi.wxs
@@ -721,7 +721,7 @@

                     <Directory Id="BINDIR" Name="bin">
                         <Component Id="bin.libcrypto_1_1.dll" Guid="{811694A9-DC99-4CFA-BD72-4877F8E32C7C}">
-                            <File Name="libcrypto-1_1$(var.OPENSSL_PLAT).dll" Source="!(bindpath.openvpn)bin\libcrypto-1_1$(var.OPENSSL_PLAT).dll Checksum="yes"/>
+                            <File Name="libcrypto-1_1$(var.OPENSSL_PLAT).dll" Source="!(bindpath.openvpn)bin\libcrypto-1_1$(var.OPENSSL_PLAT).dll" Checksum="yes"/>
                         </Component>
                         <Component Id="bin.liblzo2_2.dll" Guid="{A1A057F0-235B-4F55-9A3A-CDA2112648F8}">
                             <File Name="liblzo2-2.dll" Source="!(bindpath.openvpn)bin\liblzo2-2.dll" Checksum="yes"/>


```

**Pull request merged**